### ISSUE TITLE
Adjust CustomHypermediaType so it produces HAL-like JSON

### ISSUE THREAD

**Issue by gregturn** [spring-projects/spring-hateoas#1260]
More and more, people are using our test example, `CustomHypermediaType`, to produce HAL-like JSON, but served under a different media type.

To avoid questions like, "why doesn't this produce the right JSON", it makes sense to _properly_ register all the parts of HAL.

From a project perspective, it only alters some test cases. But from an example perspective, it should be quite handy.

**Comment by odrotbohm** [comment]
I don't think we should tweak our tests in that direction. The addition of a custom media type is more than just supporting a different name for an already supported one. By moving the custom media type used in the test closer to HAL we risk the test to stay green if we change something in the custom media type activation code in a way that loses that custom declaration but rather exposes existing HAL support.

I'll chime in on the original discussion in a moment.
